### DISCUSSION TITLE
Create a Plan Definition for the Covax App

### DISCUSSION THREAD

**Discussion by rkodev** [opensrp/fhircore#420] [Ideas]
Define a Plan Definition for the Covax App in [Hapi Server](https://www.hl7.org/fhir/plandefinition.html)

**Action Points**

- [x] Research on implementation of a plan definition in Hapi Fhir Server. An example of a plan definition supported by [Hapi Fhir Server](https://github.com/hapifhir/hapi-fhir/blob/master/example-projects/hapi-fhir-jpaserver-cds-example/src/test/resources/ca/uhn/fhir/jpa/cds/example/plandefinition-apply.json)
- [x] Create a sample PlanDefinition resource in Hapi Fhir Server
- [ ] Discussion on workflow to support PlanDefinition authoring

**Comment by fredhersch** [comment]
For Immunizations, first it would be worth establishing whether you are using a CarePlan (via a PlanDefinition) or ImmunizationRecommendation - from the FHIR Spec:

_[An ImmunizationRecommendation](https://www.hl7.org/fhir/immunizationrecommendation.html) can be interpreted as a narrow type of CarePlan dealing only with immunization events. Where such information could appear in either resource, the immunization-specific resource is preferred._

Then, I'd propose creating the static resources as sample data that could be used to:

1. Identify the activities that will be captured in the CarePlan and any associated events they will trigger
2. Validate the required elements in the CarePlan for the COVID use case
3. Work backwards to the PlanDefinition and ActivityDefinitions
4. Continue to develop other aspects of the app

**Comment by f-odhiambo** [comment]
The first part of your response seems like a research ticket i.e. Establish the creation CarePlan generation either via a PlanDefinition or ImmunizationRecommendation. 

Some sample static resources can be found [here](https://github.com/opensrp/fhircore/tree/main/docs/patient-resource/covid-resources)

**Comment by MuhammadSalman-7214** [comment]
what me and @rkodev decided uptil now is we will have multiple plan definition json files for multiple vaccines for Covid App. For example Mortana vaccine has one specific plan definition file uploaded to server, similarly for other vaccines. The vaccine user select in the application, we'll have the plan definition available for that vaccine and we'll also have patient resource available, so that will help us in creating Careplan for a specific patient and in general.

**Comment by pld** [comment]
Why multiple plan defs instead of single plan def w/multiple vaccines?

**Comment by rkodev** [comment]
@pld We are currently using 1 Plan definition and Specifying using conditions and triggers to qualify the different actions @MuhammadSalman-7214

**Comment by f-odhiambo** [comment]
**Update** 
1. PlanDefinition created by Salman, tested by Ronald and reproducing a careplan

**Comment by f-odhiambo** [comment]
Update: Plandefition for the Covax app here https://cds-sandbox.alphora.com/cqf-ruler-r4/fhir/PlanDefinition/3172. Being tested on the CDS server by Bryn for CarePlan generation
